runtime/src/main/java/org/corfudb/runtime/collections/StoreTransaction.java

```diff
@@ -0,0 +1,152 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Message;
+import org.corfudb.protocols.logprotocol.SMREntry;
+import org.corfudb.runtime.CorfuStoreMetadata;
+import org.corfudb.runtime.CorfuStoreMetadata.Timestamp;
+import org.corfudb.runtime.object.transactions.TransactionalContext;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.BiFunction;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+public interface StoreTransaction<T extends StoreTransaction<T>> {
```

Comment:
Can we avoid this parent interface all together to simplify the changes - TxnContext() is just for normal reads and writes while ScopedTransaction is used purely for snapshot reads?